easy-table/easy-table.d.ts

```diff
@@ -1,40 +1,222 @@
+
+		/**
+		 * Get list of columns in printing order
+		 *
+		 * @returns {string[]}
+		 */
+		public columns(): string[];
+
+		/**
+		 * Format just rows, i.e. print the table without headers and totals
+		 *
+		 * @returns {String} String representaion of the table
+		 */
+		public print(): string;
+
+		/**
+		 * Format the table
+		 *
+		 * @returns {String}
+		 */
+		public toString(): string;
+
+		/**
+		 * Push delimeter row to the table (with each cell filled with dashs during printing)
+		 *
+		 * @param {String[]} [cols]
+		 * @returns {Table} `this`
+		 */
+		public pushDelimeter(cols?: string[]): EasyTable;
+
+		/**
+		 * Compute all totals and yield the results to `cb`
+		 *
+		 * @param {Function} cb - Callback function with signature `(column, value, printer)`
+		 */
+		public forEachTotal<T>(cb: (column: string, value: T, printer: CellPrinter<T>) => void): void;
+
+		/**
+		 * Format the table so that each row represents column and each column represents row
+		 *
+		 * @param {IPrintColumnOptions} [opts]
+		 * @returns {String}
+		 */
+		public printTransposed<T>(opts?: IPrintColumnOptions<T>): string;
+
+		/**
+		 * Sort the table
+		 *
+		 * @param {Function|string[]} [cmp] - Either compare function or a list of columns to sort on
+		 * @returns {Table} `this`
+		 */
+		public sort(cmp?: string[]): EasyTable;
+		/**
+		 * Sort the table
+		 *
+		 * @param {Function|string[]} [cmp] - Either compare function or a list of columns to sort on
+		 * @returns {Table} `this`
+		 */
+		public sort<T>(cmp?: CompareFunction<T>): EasyTable;
+
+		/**
+		 * Add a total for the column
+		 *
+		 * @param {String} col - column name
+		 * @param {Object} [opts]
+		 * @returns {Table} `this`
+		 */
+		public total<T>(col: string, opts?: ITotalOptions<T>): EasyTable;
+		/**
+		 * Predefined helpers for totals
+		 */
+		public static aggr: IAggregators;
+
+		/**
+		 * Print the array or object
+		 *
+		 * @param {Array|Object} obj - Object to print
+		 * @param {Function|Object} [format] - Format options
+		 * @param {Function} [cb] - Table post processing and formating
+		 * @returns {String}
+		 */
+		public static print<T>(obj: T | T[], format?: FormatFunction<T> | IFormatObject, cb?: TablePostProcessing): string;
+
+		/**
+		 * Same as `Table.print()` but yields the result to `console.log()`
+		 */
+		public static log<T>(obj: T | T[], format?: FormatFunction<T> | IFormatObject, cb?: TablePostProcessing): void;
+		/**
+		 * Same as `.toString()` but yields the result to `console.log()`
+		 */
+		public log(): void;
+	}
+
+	type CellPrinter<T> = (val: T, width: number) => string;
+	type CompareFunction<T> = (a: T, b: T) => number;
+	type ReduceFunction<T> = (acc: T, val: T, idx: number, length: number) => T;
+	type FormatFunction<T> = (obj: T, cell: (name: string, val: any) => void) => void;
+	type TablePostProcessing = (result: EasyTable) => string;
+
+	interface IPrintColumnOptions<T> {
```

Comment:
please remove `I` prefix from interface.
from http://www.typescriptlang.org/Handbook#writing-dts-files 

> Naming Conventions
In general, do not prefix interfaces with I (e.g. IColor). Because the concept of an interface in TypeScript is much more broad than in C# or Java, the IFoo naming convention is not broadly useful.